### ISSUE TITLE
printf %ld regression from 20200725a to 20200731 on 32-bit big-endian platforms

### ISSUE THREAD

**Issue by zv-io** [ThomasDickey/original-mawk#69]
On 32-bit big-endian platforms (such as ppc or sparc), there was a regression in the printf `%ld` format specifier. Minimal reproducer:

```
# 32-bit ppc and sparc
$ echo 1 | awk '{printf "%d\n%ld\n", $1,$1}'
1
0
```

```
# 32-bit x86, arm, etc.
$ echo 1 | awk '{printf "%d\n%ld\n", $1,$1}'
1
1
```

More information:

  * https://git.adelielinux.org/adelie/packages/-/issues/1010
  * https://git.adelielinux.org/adelie/packages/-/issues/1006

I can provide access to a 32-bit PowerPC or SPARC environment, though [QEMU user](https://www.qemu.org/docs/master/user/main.html) may be sufficient for debugging/testing.

**Comment by ThomasDickey** [comment]
agreed - Zach van Rijn sent mail about this on April 19.  I was able to reproduce the problem, and intend working on mawk once I'm done with fixes for byacc.

**Comment by ThomasDickey** [comment]
The relevant change-log item is indeed [Debian #303825](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=303825) though the relationship could use a couple of lines for clarity.  I'll do that.  At the moment, I'm studying a (simple) fix for this problem.

**Comment by ThomasDickey** [comment]
Released fix ([20230525](https://invisible-island.net/mawk/CHANGES.html#t20230525))
